### PR TITLE
[DS-2794] Increase logging if a bean retrieval fails or if the database connection is null

### PR DESCRIPTION
**Pull request by KevinVdV**
This pull requests adds additional logging if beans cannot be retrieved, this will be particularly useful of the hibernate SessionFactory bean. If this bean fails to init the database connection in a context could be null. An example stacktrace is shown below:

Caused by: org.springframework.beans.factory.BeanCreationException: Could not autowire field: private org.hibernate.SessionFactory org.dspace.core.HibernateDBConnection.sessionFactory; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'sessionFactory' defined in file [/Users/kevin/Development/DSpace/dspace-kevin/config/spring/api/core-hibernate.xml]: Invocation of init method failed; nested exception is org.hibernate.HibernateException: Missing column: dspace_object in public.doi
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:531)

In case the database connection is null an additional log is added to provide guidance.

### CODE REVIEW

File: dspace-api/src/main/java/org/dspace/core/Context.java [hunk 1]
```diff
@@ -131,6 +131,11 @@ private void init()
         {
             // Obtain a non-auto-committing connection
             dbConnection = new DSpace().getSingletonService(DBConnection.class);
+            if(dbConnection == null)
+            {
+                log.fatal("Cannot obtain the bean the provides a database connection, " +
```
Comment (mwoodiupui) [inline dspace-api/src/main/java/org/dspace/core/Context.java]: Picky language stuff:
1.  replace the second "the" with "which".
2.  Comma splice:  replace the comma with a period and begin a new sentence below.
With those changes, or even without, +1.

Comment (KevinVdV) [inline dspace-api/src/main/java/org/dspace/core/Context.java]: Fixed the wording, thanks for the checkup.

**Comment by tdonohue** [comment]
+1 this looks like an obvious benefit to help us debug issues.

**Comment by tdonohue** [comment]
I'm going to merge this immediately, as it seems obvious / noncontroversial and will provide immediate benefits as we test/debug master

**Pull request merged**